dao/src/main/java/uk/gov/moj/sdt/dao/BulkSubmissionDao.java

```diff
@@ -69,11 +68,7 @@ public IBulkSubmission getBulkSubmission(final IBulkCustomer bulkCustomer, final
         TypedQuery<BulkSubmission> typedQuery = getEntityManager().createQuery(criteriaQuery.select(root)
                 .where(sdtCustomerPredicate, customerReferencePredicate, sdtDatePredicate));
 
-try {
-            return typedQuery.getResultStream().findFirst().orElse(null);
-        } catch(NoResultException e) {
-            return null;
-        }
+        return typedQuery.getSingleResult();
```

Comment:
What's the reasoning behind this change?  Previously, if nothing is found then null was returned, now an exception will be thrown.  At least one instance of code that calls this expects a null response (BulkSubmissionValidator.visit()), so it won't function as expected.

Reply:
call form BulkSubmissionValidator.visit() is fine, it is checking for null value already

Reply:
The JavaDoc for getSingleResult() states that it will throw a NoResultException is no result is found.  The previous version of the code caught this exception and returned null.  Instead of doing that, this version of the code throws the exception.  This will cause BulkSubmissionValidator.visit() to fail because it's set up to handle a null response rather than the exception.

This code needs to be changed so that it matches the behaviour of the previous version.

Reply:
reverted this change now